code/modules/power/apc.dm

```diff
@@ -799,6 +799,39 @@
 	. = ..()
 	if(.)
 		return
+	var/mob/living/carbon/human/H = user
+	var/datum/species/ethereal/eth_species = H.dna?.species
+	if(istype(H) && istype(eth_species))
+		if(H.a_intent == INTENT_HARM)
+			if(eth_species.ethereal_charge > 145)
+				to_chat(H, "<span class='warning'>Your charge is full!</span>")
+				return
+			to_chat(H, "<span class='notice'>You start channeling some power through the APC into your body.</span>")
+			if(do_after(user, 70, target = src))
+				if(cell.charge >= (cell.maxcharge / 2) || (eth_species.ethereal_charge >= ETHEREAL_CHARGE_FULL))
+					to_chat(H, "<span class='notice'>You receive some charge from the APC.</span>")
+					eth_species.adjust_charge(20)
```

Comment:
I don't know if 20 is a lot or not. Is 20 a lot? what's the max charge on ethereals?

Reply:
max charge on ethereals is ~150